actor-server/actor-core/src/main/scala/im/actor/server/dialog/DialogExtension.scala

```diff
@@ -57,18 +58,24 @@ final class DialogExtensionImpl(system: ActorSystem) extends DialogExtension wit
     }
 
   def sendMessage(
-    peer:          ApiPeer,
-    senderUserId:  UserId,
-    senderAuthSid: Int,
-    senderAuthId:  Option[Long], // required only in case of access hash check for private peer
-    randomId:      RandomId,
-    message:       ApiMessage,
-    accessHash:    Option[Long]   = None,
-    isFat:         Boolean        = false,
-    forUserId:     Option[UserId] = None
+    peer:                 ApiPeer,
+    senderUserId:         UserId,
+    senderAuthSid:        Int,
+    senderAuthId:         Option[Long], // required only in case of access hash check for private peer
+    randomId:             RandomId,
+    message:              ApiMessage,
+    accessHash:           Option[Long]           = None,
+    isFat:                Boolean                = false,
+    forUserId:            Option[UserId]         = None,
+    quotedHistoryMessage: Option[HistoryMessage] = None
   ): Future[SeqStateDate] =
     withValidPeer(peer.asModel, senderUserId, Future.successful(SeqStateDate())) {
+      //todo hp: send content of message and public id?
       // we don't set date here, cause actual date set inside dialog processor
+      val quotedMessage = quotedHistoryMessage flatMap (hm ⇒ ApiMessage.parseFrom(hm.messageContentData).fold(
```

Comment:
mm, I suggest one principle:
if bob reply to alice's message, bob call the SendMessage rpc with a QuotedMessageReference **with**  valid content (Text or Document ...).

bot if bob forwards alice's message  he just call SendMesage rpc with QuotedMessageReference and **EmptyMessage**
(and if bob has a comment on forwarded message, he should send a normal message by calling the sendMessage rpc (maybe before of forward) )

I mean
 Reply:
send message with valid content
Forward:
send message with empty content.